subsys/net/lib/config/init.c

```diff
@@ -32,7 +32,7 @@ extern const struct log_backend *log_backend_net_get(void);
 extern int net_init_clock_via_sntp(void);
 
 static K_SEM_DEFINE(waiter, 0, 1);
-static struct k_sem counter;
+static K_SEM_DEFINE(counter, 0, UINT_MAX);
```

Comment:
Initialize the semaphore to zero available tokens, which should match previous behavior. (...mostly. I'm pretty sure that previously this was "initialized" to a limit of zero, so e.g. `k_sem_give` calls would be silently ignored until the init, but I also suspect that was a bug...).

I don't know if this is the proper proper way to do this here - I'm inclined to say "directly initialize this to 1/UINT_MAX and get rid of the k_sem_init call below", but I don't know the networking subsystem at all.

Reply:
Also, TIL that any polling APIs on a zero-initialized struct will null-pointer-dereference.